src/lib/parser/default_impl_populator.rs

```diff
@@ -0,0 +1,32 @@
+use std::collections::BTreeMap;
+
+use crate::{ast::visit_mut::*, ast::*};
+
+#[derive(Debug)]
+pub struct  DefaultImplPopulator {
+    pub traits: BTreeMap<String, Trait>,
+}
+
+impl<'a> VisitorMut<'a> for DefaultImplPopulator {
+    fn visit_trait(&mut self, trait_: &mut Trait) {
+        self.traits.insert(trait_.name.get_name(), trait_.clone());
```

Comment:
This does not respect the scope separation, just like everywhere else. Traits resolution must always be localized.

Reply:
This is a change beyond the scope of this PR, need a new issue for that